cmp/form/HoistField.js

```diff
@@ -32,20 +33,26 @@ import {observable, setter, computed} from 'hoist/mobx';
  * control, so that changes to its value do not cause the parent of this
  * control to re-render.
  *
- * Hoist Fields generally support the following properties.
- *
- * @prop value, string
- * @prop onChange, handler to fire when value changes
- * @prop onCommit, handler to fire when value is committed.
- *
- * @prop model, model to bind to
- * @prop field, name of property in model to bind to
- * @prop disabled, is control disabled
- * @prop style
- * @prop className
+ * Hoist Fields generally support the properties documented below.
  */
 export class HoistField extends Component {
 
+    static propTypes = {
+        value: PT.string,
+        /** handler to fire when value changes */
+        onChange: PT.func,
+        /** handler to fire when value is committed */
+        onCommit: PT.func,
+        /** model to bind to */
+        model: PT.object,
+        /** name of property in model to bind to */
+        field: PT.string,
+        /** is control disabled */
+        disabled: PT.bool,
```

Comment:
For the callbacks I would communicate briefly that they get passed the new value. Could skip disabled as self-evident. Would pref caps.

Reply:
Are these prop types actually checked on the subclass instances?  Or are they overridden by the specification of propTypes on the subclass?   The latter seems more likely to me.
 Would be good to know....

Reply:
It appears that they are overridden by the subclass. Passed a className prop to a dayField as a int. The propType check on Hoist field did not pick it up. I had to add the PropType definition for className to dayField specifically to catch it.

Reply:
From Lee: Perhaps we just want to have a single block of these on
HoistField? Where we indicate the optional ones that only apply to certain
controls at the bottom, with the controls they apply to?

Reply:
I think that sounds great in terms of functionality and avoids having to redefined the ones common to HoistField if we want the check on the subclasses. 

However, it would take away the convenience of see subclass specific props at the top of that class (unless we replace the doc string which we were moving away from)

Reply:
From Lee:

Something has to give here. We need to revisit HoistField in a number of
ways. Let's leave them as is (but in a consistent state, with all shared
ones on HoistField only) and address later.